core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java

```diff
@@ -82,25 +116,26 @@ public void testBulkUpdateSimple() throws Exception {
             assertThat(bulkItemResponse.getIndex(), equalTo("test"));
         }
 
+        final Script script = new Script("ctx._source.field += 1", ScriptType.INLINE, CustomScriptPlugin.NAME, null);
+
         bulkResponse = client().prepareBulk()
-                .add(client().prepareUpdate().setIndex(indexOrAlias()).setType("type1").setId("1")
-                        .setScript(new Script("ctx._source.field += 1", ScriptService.ScriptType.INLINE, null, null)))
-                .add(client().prepareUpdate().setIndex(indexOrAlias()).setType("type1").setId("2")
-                        .setScript(new Script("ctx._source.field += 1", ScriptService.ScriptType.INLINE, null, null)).setRetryOnConflict(3))
+                .add(client().prepareUpdate().setIndex(indexOrAlias()).setType("type1").setId("1").setScript(script))
+                .add(client().prepareUpdate().setIndex(indexOrAlias()).setType("type1").setId("2").setScript(script).setRetryOnConflict(3))
                 .add(client().prepareUpdate().setIndex(indexOrAlias()).setType("type1").setId("3")
-                        .setDoc(jsonBuilder().startObject().field("field1", "test").endObject())).execute().actionGet();
+                        .setDoc(jsonBuilder().startObject().field("field1", "test").endObject()))
+                .get();
 
         assertThat(bulkResponse.hasFailures(), equalTo(false));
         assertThat(bulkResponse.getItems().length, equalTo(3));
         for (BulkItemResponse bulkItemResponse : bulkResponse) {
             assertThat(bulkItemResponse.getIndex(), equalTo("test"));
         }
-        assertThat(((UpdateResponse) bulkResponse.getItems()[0].getResponse()).getId(), equalTo("1"));
```

Comment:
I totally agree with your comment but I don't know any way to check for redundant casts like Intellij is doing.